### PR TITLE
GameDB: Add fixes to various games

### LINKED ISSUE #964: [PAL] Ratchet and Clank 2 Going Commando crashes after clearing Jammer Array level
It dumps this into the console when run without any speed hacks.

```
EE: Unrecognized MMI op 700000a5
EE: Unrecognized op 1108401
EE: Unrecognized op 3e83001
EE: Unrecognized op 1efd0bd
EE: Unrecognized op 1f63001
EE: Unrecognized op 1f4e8bd
EE: Unrecognized op 1eeb0bd
EE: Unrecognized op 1eeb0bd
EE: Unrecognized op 1eda9bd
EE: Unrecognized op 1108401
EE: Unrecognized op 3e83001
EE: Unrecognized op 1efd0bd
EE: Unrecognized op 1f63001
EE: Unrecognized op 1f4e8bd
EE: Unrecognized op 1eeb0bd
EE: Unrecognized op 1eeb0bd
EE: Unrecognized op 1eda9bd
EE: Unrecognized op 1108401
EE: Unrecognized op 3e83001
EE: Unrecognized op 1efd0bd
EE: Unrecognized COP0 op 41800000
EE: Unrecognized op 1e0c61e
EE: Unrecognized op 1fa3001
EE: Unrecognized op 1f4e8bd
EE: Unrecognized op 1f63801
```

I've also seen it throw this exception during several of the runs
    /home/hirato/sources/pcsx2-git/src/pcsx2/pcsx2/FiFo.cpp(47) : assertion failed:
        Function:  void ReadFIFO_VIF1(mem128_t*)
        Thread:    EE Core
        Condition: vif1Regs.stat.FQC != 0
        Message:   FQC = 0 on VIF FIFO READ!

```
Stacktrace:
[00] wxNavigationEnabled<wxControl>::AcceptsFocusRecursively() const
[01] wxMDIParentFrameBase::TryBefore(wxEvent&)   
[02] 0x0x9fce112                                 
[03] 0x0x9f4b05f                                 
[04] wxNavigationEnabled<wxControl>::AcceptsFocusRecursively() const
[05] wxString wxString::Format<char const*>(wxFormatString const&, char const*)
[06] wxNavigationEnabled<wxControl>::AcceptsFocusRecursively() const
[07] wxNavigationEnabled<wxControl>::AcceptsFocusRecursively() const
[08] wxNavigationEnabled<wxControl>::AcceptsFocusRecursively() const
[09] 0x0xf64aa315                                
[10] clone                                       
```

the build script I'm using it outputting pcsx2-git-1.3.1.r2301.ac0e40a for the current build I'm using.

EDIT

https://gist.github.com/Hirato/86d846d6f6a362f6358a The console log is here.

It's not reliable though, sometimes it has the TLB miss section, sometimes not.
Sometimes the unrecognised opcodes are two instances of '1'
Sometimes it crashes and at other times it just leaves you with a black screen from which you can reboot or restore an earlier save state. 

What happens is that after clearing the Jamming Array level, a cutscene begins to play; The screen is black, the GS is seemingly doing nothing and the errors then pop up in the console.

No combinations of EE settings I've tried sidesteps the issue, ditto for any combination of speedhacks. I haven't messed around much with the plugin settings, but it also happens in software mode.

plugins are GSdx-OGL, onepad, and spu2x, rest are null drivers.

the ISO is one I ripped myself using the `dd` tool which plays fine on actual PS2 hardware, so I do not think the iso itself is broken.

### LINKED ISSUE #1976: GameDB improvement and missing patch files
image

### PR DESCRIPTION
**Pull request by lightningterror**
- Add fix for R&C2 Going Commando - Fixes #964
fixes cinematic and crash  
- Add fix for Metal Gear Solid 3: Subsistence  
 fixes screen artifacts
- Add multiple fixes for Max Payne  
- Add / correct fix for Mercenaries 2  
Corrected incorrect title and added timing fix

Add ScarfaceIbitHack in Gameindex since now it will be used

Thanks to @atomic83GitHub for testing out the games.
Close #1976

Code tidying up

### CODE REVIEW

**Comment by lightningterror** [comment]
I'm not sure If I should add the gamefixes for other versions of Metal Gear Solid 3 - Subsistence.
I'll wait for feedback.

**Comment by atomic83GitHub** [comment]
God décision lets see.

File: bin/GameIndex.dbf [hunk 1]
```diff
@@ -40213,6 +40242,12 @@ Name   = Metal Gear Solid 3 - Subsistence [Disc1of3]
 Region = NTSC-U
 Compat = 5
 ---------------------------------------------
+Serial = SLUS-21243
```
Comment (MrCK1) [inline bin/GameIndex.dbf]: This shouldn't be moved, the serials are in order numerically, not alphabetically or grouped by game.

File: bin/GameIndex.dbf [hunk 2]
```diff
@@ -17881,6 +17907,11 @@ Name   = Metal Gear Solid 3 - Subsistence [Disc2of3]
 Region = PAL-I
 MemCardFilter = SLES-82044
 ---------------------------------------------
+Serial = SLES-82051
```
Comment (MrCK1) [inline bin/GameIndex.dbf]: Ditto.

**Comment by atomic83GitHub** [comment]
. For Max Payne , these fix avoid random crash and improve speed and stability of the game.
. For Rayman Revolution , the EE Timing Hack avoid some ingame error cutsceen

**Comment by atomic83GitHub** [comment]
I think it will be beter to added the MGS3 to all version for the accuracy and to prevent bugs

File: bin/GameIndex.dbf (hunk 1 above)
Comment (lightningterror) [inline bin/GameIndex.dbf]: Ah I didn't know that , I thought they were alphabetically , well at least some.
ocd gets in the way xD

**Comment by atomic83GitHub** [comment]
There is à error in the max payne patch (uClampmode)

**Comment by atomic83GitHub** [comment]
This must be changed To vuClampMode

**Comment by atomic83GitHub** [comment]
And this error was not present when i had test Max Payne , must be corrected now.

**Comment by lightningterror** [comment]
> eeRoundMode = 0
> eeClampMode = 3
> vuRoundMode = 0
> vuClampMode = 3
> EETimingHack = 1
> ScarfaceIbitHack = 1

I need to know what each fix changes in Max Payne. Are all of them really needed?

**Comment by atomic83GitHub** [comment]
Yes . Vu round mode and clamping avoid random VUs error . Ee clamping and round mode avoid tlb miss and some ee error. (Yes this game was very buggy) .EE Timing hack avoid core issues (same issues than ratchet and clank 2).And I bit hack improve speed and stability.

**Comment by atomic83GitHub** [comment]
All of these are needed for à Perfect experience.

**Comment by atomic83GitHub** [comment]
It would be nice to added the Vu I bit hack to the game : Scarface the world is yours , and the ffx video fix is missing from the database (added it to the all final fantasy 10 games) also , Disney Treasure planet pal M4 dont have the hacks added to the other version (EETimingHack,eeRoundMode = 0,eeClampMode = 3).

**Comment by atomic83GitHub** [comment]
Oups final fantasy 10 also have the IPUwait hack (wich is the ffx video fix) sorry.But for Scarface and Disney Treasure planet , its real.

**Comment by atomic83GitHub** [comment]
To resume , these are the final improvment to achieve the topic:
Add the Vu I bit hack to Scarface the world is your's
Add the EETimingHack , eeRoundMode = 0 ,  eeClampMode = 3 to the pal m4 version of Disney Treasure Planet 
Add eeClampMode = 2 (extra) , vuClampMode = 2 (extra) to all Final Fantasy 10 versions (avoid crash) in addition to the IPUwaithack

**Comment by lightningterror** [comment]
Serials please.

Also @MrCK1 is it ok to add fixes game versions which haven't been tested or should I stick to tested only?

**Comment by atomic83GitHub** [comment]
For Scarface its : SLUS-21111 , SLES-54182 , SLES-54183 , SLES-54184 , SLES-54271
For Disney Treasure its : SCES-51176
For Final Fantasy its : SLPM-67513 , SLPS-25050 , SLPS-25088 , SLPS-72501 , SLUS-20312 , SCED-50642 , SCED-50907 , SCES-50490 , SCES-50491 , SCES-50492 , SCES-50493 , SCES-50494 , SLPM-65115 , SLPM-66124 , SLPM-66677

**Comment by atomic83GitHub** [comment]
I have forget for Final Fantasy 10 to change the EE/IOP RoundMode to nearest for the backwards bosses faces.

**Comment by MrCK1** [comment]
@lightningterror tested only, we don't want to assume all regions need the same gamefix.

**Comment by refractionpcsx2** [comment]
I'm surprised the scarface hack works for Max Payne, that was a very specific hack.

**Comment by atomic83GitHub** [comment]
#refractionpcsx2 , yes but the issues are very similar and enebeled it improve speed and stability to the game (like the TLB preload hack in certains demo disc).

**Comment by atomic83GitHub** [comment]
I had check the GameDB and I had notice this morning , there are missing entries :
SLES 55031 : Kung Fu Panda: The Game (add the FMVinsoftwareHack , i had tested it earlier and imposible the see the right part of all cinematics with the memory wraping or unscale point and line)
SLES 54905 : The Simpsons Game (No patch needed)

**Comment by atomic83GitHub** [comment]
There is an error in my remarks, ratchet and clank 2 crash at diferents areas with the timing hack, this hack must be removed immediately: sorry I think a custom patch is needed for it.

**Comment by atomic83GitHub** [comment]
Its better to manualy enebeled it.

**Comment by atomic83GitHub** [comment]
Same for Rayman Revolution but other games are good.

**Comment by atomic83GitHub** [comment]
All seems to be good

**Comment by lightningterror** [comment]
> For Scarface its : SLUS-21111 , SLES-54182 , SLES-54183 , SLES-54184 , SLES-54271
> For Disney Treasure its : SCES-51176
> For Final Fantasy its : SLPM-67513 , SLPS-25050 , SLPS-25088 , SLPS-72501 , SLUS-20312 , SCED-50642 , SCED-50907 , SCES-50490 , SCES-50491 , SCES-50492 , SCES-50493 , SCES-50494 , SLPM-65115 , SLPM-66124 , SLPM-66677

I need actual confirmation on games that you can actually test. 
Like MrCK1 said we don't want to assume all regions need the same gamefix.

**Comment by atomic83GitHub** [comment]
I had tested the disney treasure planet and its works prety good , for Scarface I did not have the game but the patch was crafted for this game so It would be nice to added the patch to the game.
And for final fantasy 10 wee  should not add patch if wee cant test.

**Comment by atomic83GitHub** [comment]
After some reflections , I think its better to remove the timing hack to Rayman Revolution.

**Comment by refractionpcsx2** [comment]
It was crafted for one version of the game, can you make sure all versions are tested by somebody please.

**Comment by atomic83GitHub** [comment]
No I can't , so I think its better to only apply it to pal versions.
Also , please remove the Rayman Revolution Timing Hack just for safety because I never finish the entire game with it.

**Comment by atomic83GitHub** [comment]
Is possible that someone send the actual GameDB for a control of improvement ?

**Comment by lightningterror** [comment]
> Is possible that someone send the actual GameDB for a control of improvement ?

You have it. GameIndex.dbf
The file is the same used in the source code so it doesn't need any compiling.
All the instructions are inside as well.

**Comment by atomic83GitHub** [comment]
Lets do the report.
I have completed every game at the top to the bottom and tested the stability of every game (especialy Max Payne) so , I will indicate the final improvement and minor removes/updates , let's do this:
---------------------------------------------------
Mercenaries 2 : corrected the frensh issue and the missing patch (already done)

Max Payne : add Multiple hack to avoid some crash and error (Need update!#)
#Remove the Vu I bit hack to the database to avoid stutering in some levels.

Ratchet and Clank 2 (Pal versions only!) add the TimingHack to avoid crash in some ingame cinematics (already done)

Disney treasure planet : add the hacks for the Pal M4 version (Must be update)
EETimingHack , eeRoundMode = 0 , eeClampMode = 3

Scarface The World is your's : add the Vu I bit hack to all Pal versions (other not tested) (Must be update)

Rayman revolution : Remove the Timing hack to the database for safety (Need update!)

Metal gear solid 3 subsistance : Add the VUs Clamping mode to extra + preserv sign to avoid screen artifact (already done)
----------------------------------------------------
Add missing entries:
SLES 55031 : Kung Fu Panda: The Game (No patch needed)
SLES 54905 : The Simpsons Game (No patch needed)
SLES 54734 : Ratatouille (No patch needed)
----------------------------------------------------
This is finish , now we can change/add some value before integrate the PR to PCSX2
Hope get you help.

**Comment by atomic83GitHub** [comment]
Thanks , I will test and (probably aprove :) ) the new GameDB

**Comment by atomic83GitHub** [comment]
I confirm , the new database is perfect , we can introduce it to pcsx2.

**Comment by lightningterror** [comment]
Well seems the travis build is stuck.

**Comment by Hirato** [comment]
The EE-Timing hack is only needed for R&C2 in a single cutscene after the Jammer Array level; everything else works without it.
I'd appreciate if you could amend the comment to say such so it's more obvious to people who look at it in the future. eg `Fixes crash & cinematic after Jammer Array level`

And when this is merge, I'd much rather that #964 not be closed until the underlying issue has been identified and fixed; I'm sure it's something that affects a lot of games.

**Comment by gregory38** [comment]
Is it finish ? Or WIP ?

**Comment by lightningterror** [comment]
@gregory38 Now it's ready.
Btw I'm away from home so I can't squash commits. Accept collaborator invitation if you like and squash them for me :)

**Comment by atomic83GitHub** [comment]
ALERT! I have found a regression in metal gear solid 3 , must remove the fix , sorry , I had tested this game in software mode and the regression didn't show in this mode.

**Pull request merged**